plugin-parser/mlops-plugin-parser.cc

```diff
@@ -0,0 +1,349 @@
+          const gchar *clear = json_object_get_string_member (object, "clear");
+
+          if (!name || !model) {
+            _E ("Failed to get name or model from json file '%s'.", json_file);
+            continue;
+          }
+
+          guint version;
+          bool active = (activate && g_ascii_strcasecmp (activate, "true") == 0);
+          bool clear_old = (clear && g_ascii_strcasecmp (clear, "true") == 0);
+
+          /* Remove old model from database. */
+          if (clear_old) {
+            /* Ignore error case. */
+            ml_agent_model_delete (name, 0U);
+          }
+
+          ret = ml_agent_model_register (name, model, active, desc ? desc : "",
+              app_info ? app_info : "", &version);
+
+          if (ret == 0)
+            _I ("The model with name '%s' is registered as version '%u'.", name, version);
+          else
+            _E ("Failed to register the model with name '%s'.", name);
+        }
+        break;
+      case MLSVC_JSON_PIPELINE:
+        {
+          const gchar *name = json_object_get_string_member (object, "name");
+          const gchar *desc = json_object_get_string_member (object, "description");
+
+          if (!name || !desc) {
+            _E ("Failed to get name or description from json file '%s'.", json_file);
+            continue;
+          }
+
+          ret = ml_agent_pipeline_set_description (name, desc);
+
+          if (ret == 0)
+            _I ("The pipeline description with name '%s' is registered.", name);
+          else
+            _E ("Failed to register pipeline with name '%s'.", name);
+        }
+        break;
+      case MLSVC_JSON_RESOURCE:
+        {
+          const gchar *name = json_object_get_string_member (object, "name");
+          const gchar *path = json_object_get_string_member (object, "path");
+          const gchar *desc = json_object_get_string_member (object, "description");
+          const gchar *clear = json_object_get_string_member (object, "clear");
+
+          if (!name || !path) {
+            _E ("Failed to get name or path from json file '%s'.", json_file);
+            continue;
+          }
+
+          bool clear_old = (clear && g_ascii_strcasecmp (clear, "true") == 0);
+
+          /* Remove old resource from database. */
+          if (clear_old) {
+            /* Ignore error case. */
+            ml_agent_resource_delete (name);
+          }
+
+          ret = ml_agent_resource_add (
+              name, path, desc ? desc : "", app_info ? app_info : "");
+
+          if (ret == 0)
+            _I ("The resource with name '%s' is registered.", name);
+          else
+            _E ("Failed to register the resource with name '%s'.", name);
+        }
+        break;
+      default:
+        _E ("Unknown data type '%d', internal error?", json_type);
+        break;
+    }
+  }
+}
+
+extern "C" int
+PKGMGR_MDPARSER_PLUGIN_INSTALL (const char *pkgid, const char *appid, GList *metadata)
```

Comment:
IMO need '@brief' tag for all functions.

Reply:
I added those tags. PTAL